libibumad/umad.h

```diff
@@ -183,7 +183,7 @@ int umad_status(void *umad);
 ib_mad_addr_t *umad_get_mad_addr(void *umad);
 int umad_set_grh_net(void *umad, void *mad_addr);
 int umad_set_grh(void *umad, void *mad_addr);
-int umad_set_addr_net(void *umad, int dlid, int dqp, int sl, int qkey);
+int umad_set_addr_net(void *umad, __be16 dlid, __be32 dqp, int sl, __be32 qkey);
```

Comment:
There is  a slight chance, it changes the underlying types a tiny bit, notably swapping in a short..

However, I am not aware of any architecture that would generate different machine code for this case..

If there is doubt we can symbol version it..